src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -322,6 +338,15 @@ private async Task ProcessInvAsync(INetworkPeer peer, InvPayload invPayload)
                     continue;
                 }
 
+                lock (this.processedTransactionsLockObject)
+                {
+                    // If we already processed the transaction in any of the mempool behaviors, we don't need to ask for it again.
+                    if (this.processedTransactions.Contains(inv.Hash))
```

Comment:
Is this giving us more than what `this.orphans.AlreadyHaveAsync` does?

Reply:
Yes, it aims to ensure to not download already processed transactions, the orphan concept is different: 

https://github.com/stratisproject/StratisBitcoinFullNode/issues/3956#issuecomment-532761979

> Orphans are trasnactions that we don't know the input for, if a trx is spending an unconfirmed input and for some reason the child arrived before the parent it will go in to orphans, when the parent arrives it will then from orphans in to the mempool.

Reply:
The method is admittedly not well-named, but it does check for the node's knowledge of non-orphan transactions too. For example, those that have recently been rejected due to validation failures, as well as those currently present in the mempool itself.

Reply:
I see. If that's true, this is a non-issue and is already implemented. Closing. (Note if you ever end up needing to lower the bandwidth usage, you can still lower it by remembering the invs, as I described it in the "Complication" part of this PR.

Reply:
Ideally, if time permits I would attempt to implement something along the lines of the erlay protocol ( https://telaviv2019.scalingbitcoin.org/files/bandwidth-efficient-transaction-relay-for-bitcoin.pdf ). That would appear to be the current state of the art regarding mempool set reconciliation.